ARCs/arc-0015.md

```diff
@@ -0,0 +1,241 @@
+---
+arc: 15
+title: Encrypted Short Messages
+description: Scheme for encryption/decryption that allows for private messages.
+author: Stéphane Barroso (@sudoweezy), Paweł Pierścionek (@urtho)
+discussions-to: https://github.com/algorandfoundation/ARCs/issues/*
+status: Draft
+type: Standards Track
+category: ARC
+created: 2022-11-21
+requires: 4
+---
+
+# Encrypted Short Messages 
+
+## Abstract
+
+The goal of this convention is to have a standard way for block explorers, wallets, exchanges, marketplaces, and more generally, client software to send, read & delete short encrypted messages.
+
+## Specification
+The key words "**MUST**", "**MUST NOT**", "**REQUIRED**", "**SHALL**", "**SHALL NOT**", "**SHOULD**", "**SHOULD NOT**", "**RECOMMENDED**", "**MAY**", and "**OPTIONAL**" in this document are to be interpreted as described in <a href="https://www.ietf.org/rfc/rfc2119.txt">RFC-2119</a>.
+
+> Comments like this are non-normative.
+
+### Account's message Application
+
+To receive a message, an Account **MUST** create an application that follows this convention:
+
+- A Local State named `public_key` **MUST** contain an `NACL Public Key (Curve 25519)`
```

Comment:
What format are you considering?
It looks like you are using base64, which is not optimal storage-wise as raw bytes would be shorter.

Is it local state or global state?

Also, I would recommend adding some versioning to allow new cryptosystems.
Maybe: a key `arc` which contains `arc15-nacl-curve25519`?

Reply:
I modified the implementation to use RawEncoding instead of base64.
Added also the `arc` key